app/connector/google-sheets/src/main/java/org/apache/camel/component/google/sheets/stream/GoogleSheetsStreamComponent.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.google.sheets.stream;
+
+import com.google.api.services.sheets.v4.Sheets;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.google.sheets.BatchGoogleSheetsClientFactory;
+import org.apache.camel.component.google.sheets.GoogleSheetsClientFactory;
+import org.apache.camel.component.google.sheets.GoogleSheetsVerifierExtension;
+import org.apache.camel.impl.DefaultComponent;
+import org.apache.camel.spi.Metadata;
+
+import java.util.Map;
+
+/**
+ * Represents the component that manages {@link GoogleSheetsStreamEndpoint}.
+ */
+@Metadata(label = "verifiers", enums = "parameters,connectivity")
+public class GoogleSheetsStreamComponent extends DefaultComponent {
+
+    @Metadata(label = "advanced")
+    private Sheets client;
+    @Metadata(label = "advanced")
+    private GoogleSheetsClientFactory clientFactory;
+    @Metadata
+    private GoogleSheetsStreamConfiguration configuration;
+
+    public GoogleSheetsStreamComponent() {
+        this(null);
+    }
+
+    public GoogleSheetsStreamComponent(CamelContext context) {
+        super(context);
+        registerExtension(new GoogleSheetsVerifierExtension("google-sheets-stream", context));
+        this.configuration = new GoogleSheetsStreamConfiguration();
+    }
+
+    public Sheets getClient(GoogleSheetsStreamConfiguration endpointConfiguration) {
+        if (client == null) {
+            client = getClientFactory().makeClient(endpointConfiguration.getClientId(), endpointConfiguration.getClientSecret(),
+                                                    endpointConfiguration.getApplicationName(), endpointConfiguration.getRefreshToken(),
+                                                    endpointConfiguration.getAccessToken());
```

Comment:
If possible I'd like to reduce line length a bit. 

This is not of course an issue but it makes it easier to review the code on GitHub as it does not display the whole line so we may miss some bits.

Reply:
totally agreed